src/pages/index.js

```diff
@@ -1,122 +1,197 @@
-const addCardFormElement = cardAddModal.querySelector(".modal__form");
-//cards input
-
-//image modal buttons
-const config = {
-  formSelector: ".modal__form",
-  inputSelector: ".modal__input",
-  submitButtonSelector: ".modal__button",
-  inactiveButtonClass: "modal__button_disabled",
-  inputErrorClass: "modal__input_type_error",
-  errorClass: "modal__error_visible",
-};
-
-const editFormValidator = new FormValidator(config, profileEditForm);
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "f1dab6ed-e5ba-44ef-bef1-7369d7e9bf0d",
+    "Content-Type": "application/json",
+  },
+});
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.setUserAvatar(res);
+  })
+  .catch((err) => {
+    console.error(`Error ${err}`);
+  });
+api
+  .getInitialCards()
+  .then((res) => {
+    cardSection.renderItems(res);
+  })
+  .catch((err) => {
+    console.error(`Error ${err}`);
+  });
+
+const cardDeleteModal = new PopupQuestion("#delete-card-modal");
+
+const editFormValidator = new FormValidator(
+  constants.config,
+  constants.profileEditForm
+);
 editFormValidator.enableValidation();
-const addFormValidator = new FormValidator(config, addCardFormElement);
+
+const addFormValidator = new FormValidator(
+  constants.config,
+  constants.addCardFormElement
+);
 addFormValidator.enableValidation();
 
+const profileAvatarFormValidator = new FormValidator(
+  constants.config,
+  constants.profileAvatarModalForm
+);
+profileAvatarFormValidator.enableValidation();
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
-const cardPopup = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+const cardPopup = new PopupWithForm(
+  "#add-card-modal",
+  // handleAddCardFormSubmit
+  (link, button, close) => {
+    api
+      .addCard(link)
+      .then((res) => {
+        //  button.textContent = "Saving ...";
+        cardPopup.renderLoading(true);
+        cardSection.addItem(res);
+      })
+
+      .catch((err) => {
+        console.error(`Error ${err}`);
+      })
+      .finally(() => {
+        cardPopup.renderLoading(false);
+        addFormValidator.toggleButtonState();
+        close;
+      });
+  }
+);
 cardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const userInfoPopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileFormSubmit
-);
-const cardSection = new Section(initialCards, renderCard, cardListEl);
-cardSection.renderItems();
+const profilePopup = new PopupWithForm("#profile-avatar-modal", (url) => {
+  // handles the avatar picture submit
+  api
+    .setUserAvatar(url)
+    .then((res) => {
+      userInfo.setUserAvatar(res);
+    })
+    .catch((err) => {
+      console.error(`Error ${err}`);
+    });
+  profileAvatarFormValidator.toggleButtonState();
+  profilePopup.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well